cpp/src/arrow/ipc/test_common.h

```diff
@@ -161,6 +162,11 @@ Status MakeUuid(std::shared_ptr<RecordBatch>* out);
 ARROW_TESTING_EXPORT
 Status MakeDictExtension(std::shared_ptr<RecordBatch>* out);
 
+ARROW_TESTING_EXPORT
+Status MakeRandomTensor(const std::shared_ptr<DataType>& type,
+                        const std::vector<int64_t>& shape, bool row_major_p,
+                        std::shared_ptr<Tensor>* out, uint32_t seed = 0);
```

Comment:
Two things:
1) Why not put this in `arrow/testing/random.h`?
2) Return `Result<Tensor>` instead?

Reply:
Because I followed `MakeRandomArray`.  It is in `arrow/ipc/test_common.h`.